controller/tap/server.go

```diff
@@ -468,20 +470,29 @@ func NewServer(
 		return nil, nil, err
 	}
 
-	s, _ := newGRPCTapServer(tapPort, controllerNamespace, k8sAPI)
+	globalConfig, err := config.Global(pkgK8s.MountPathGlobalConfig)
```

Comment:
We can consider promoting this to `controller/cmd/tap/main.go` (like what you did for `controller/cmd/public-api/main.go`), where `NewGrpcTapServer()` is called.

Reply:
Agree. Passing configs can make testing easier and unifies the behaviour. Two good things.